src/lens.c

```diff
@@ -1664,7 +1668,8 @@ static void rtn_rules(struct rtn *rtn, struct lens *l) {
     case L_STAR: {
         /* cfg(l*, N) -> N := N . N' | eps */
         struct state *s = add_state(prod);
-        RTN_BAIL(rtn);
+        if (s == NULL)
+            goto error;
```

Comment:
IMHO this changes behavior. Can you please describe why it makes sense to replace RTN_BAIL(rtn) with checking s for NULL?

Reply:
Same as for 1649 above